overlord/devicestate/devicestate.go

```diff
@@ -351,7 +352,7 @@ func remodelTasks(st *state.State, current, new *asserts.Model, deviceCtx snapst
 		_, err := snapstate.CurrentInfo(st, snapName)
 		// If the snap is not installed we need to install it now.
 		if _, ok := err.(*snap.NotInstalledError); ok {
-			ts, err := snapstateInstallWithDeviceContext(st, snapName, nil, userID, snapstate.Flags{Required: true}, deviceCtx)
+			ts, err := snapstateInstallWithDeviceContext(context.TODO(), st, snapName, nil, userID, snapstate.Flags{Required: true}, deviceCtx)
```

Comment:
remodelTasks should take the context from outside I think

Reply:
We can do this now (http://paste.ubuntu.com/p/57CgcQ3chp/) and I'm happy to push this. But unless I miss something it won't add much at this point. The snapstate.Remodel() function does not take a context yet and it does not currently need one. So we might add the context once we have a use-case. But either way is fine with me.